### ISSUE TITLE
Artifact references aren't being imported at first import

### ISSUE THREAD

**Issue by 4u7** [JetBrains/gradle-idea-ext-plugin#77]
Consider following sample:
```Kotlin
    create("kotlin-jps-plugin.jar") {
        archive("kotlin-jps-plugin.jar") {
        	...
        }
    }

    create("ideaPlugin") {
        directory("Kotlin") {

            directory("lib") {

                directory("jps") {
                	artifact("kotlin-jps-plugin.jar") // Is not present after first import
                }
            }
        }
    }
```
Reference `artifact("kotlin-jps-plugin.jar")` is not imported on first import, there is empty `jps` directory in the Project Structure view. However on second import artifact is present.

Plugin v0.5, Idea 191

**Comment by 4u7** [comment]
Looks like the problem is in `NamedDomainObjectCollection<TopLevelArtifact>` which orders its elements by name. Inside the resulting artifact collection `ideaPlugin` goes before `kotlin-jps-plugin.jar`

https://docs.gradle.org/current/javadoc/org/gradle/api/NamedDomainObjectContainer.html

**Comment by nskvortsov** [comment]
Scheduled to be fixed in 2019.1.2
